src/rust/ide/view/graph-editor/src/component/node.rs

```diff
@@ -570,6 +580,13 @@ impl Node {
             eval bg_color_anim.value ((c)
                 model.main_area.shape.bg_color.set(color::Rgba::from(c).into())
             );
+
+
+            // === VCS Handling ===
+            model.vcs_indicator.frp.set_status <+ frp.set_vcs_status.unwrap();
```

Comment:
Why unwrap? I mean, this prevents us from disabling the vcs status there, right?

Reply:
Oh I get it, we show and hide the status in next line. I think we should redesign this api towards handling the visibility of the status by the status indicator. So, I'd refactor all these 3 FRP lines to:

```rust
model.vcs_indicator.frp.set_status <- frp.set_vcs_status.unwrap();
```

and handle the visibility inside of vcs_indicator instead

Reply:
The idea here is that there is a difference between "has no status" and "has a status: it was unchanged". Right now we can easily add a new status for `unchanged` without much effort. If we change this as proposed, and we decide we want to show some notification for `unchanged` we need to do much more refactoring. 

Also, the point of the logic here is to completely remove the status indicator from the component tree if it is not needed. So instead of having some invisible shape, we remove all the shapes completely, which logically needs to be done by the node to get rid of the whole VCS component if it is not needed.